performance/plugins/testRunner.ts

```diff
@@ -4,231 +4,121 @@ import { BN } from "@polkadot/util";
-                  JSON.stringify(status.toHuman()!)
-              );
-              return;
-            }
-          })
-          .catch((err: any) => {
-            logLine(
-              testParams.logFile,
-              "\n" +
-                new Date().toUTCString() +
-                "- ERROR - " +
-                JSON.stringify(err)
-            );
-            testLog.getLog().warn(err);
-            resolve([i, 0]);
-            return -1;
-          });
-      } catch (error) {
-        logLine(
-          testParams.logFile,
-          "\n" + new Date().toUTCString() + "- ERROR - " + JSON.stringify(error)
-        );
-        testLog.getLog().warn(error);
-        resolve([i, 0]);
-      }
-    });
-  const nodeTxs = preSetupThreads[nodeIdx];
-  const indexArray = nodeTxs.map((_, index) => {
-    return index;
-  });
-  testLog
-    .getLog()
-    .info(
-      `Sending  in ${nodeThreads} Threads ${preSetupThreads[0].length} Txs...`
-    );
-  await asyncPool(nodeThreads, indexArray, runNodeTxs);
-}
-async function runTxsInBurstMode(
-  preSetupThreads: SubmittableExtrinsic<"promise", SubmittableResult>[][],
-  nodeIdx: number,
-  testParams: TestParams
-) {
-  const sorted = preSetupThreads[nodeIdx].sort(function (a, b) {
-    return (
-      parseFloat(JSON.parse(a.toString()).signature.nonce) -
-      parseFloat(JSON.parse(b.toString()).signature.nonce)
-    );
-  });
+  const limit = Math.ceil(testParams.totalTx / testParams.threads);
+  const load = Math.min(
+    limit,
+    Math.ceil(testParams.pending / testParams.threads)
+  );
+  console.info(
+    `Worker [${threadId}] starting execution thread load: ${load} limit: ${limit}`
+  );
 
-  const runNodeTxs = (i: number) =>
-    new Promise<[number, number]>(async (resolve) => {
-      const transaction = await sorted[i];
-      await transaction
-        .send(({ status }) => {
-          if (status.isBroadcast || status.isFuture || status.isReady) {
-            // eslint-disable-next-line no-console
-            console.info(
-              "Sending Tx with nonce -> " +
-                JSON.parse(sorted[i].toString()).signature.nonce
-            );
-            resolve([i, 1]);
-            return;
+  const myTxs = new Set();
+  const api = await sdk.getApi()!;
+  let exec_counter = 0;
+  let nonce_offset = 0;
+  const promises = [];
+  const now = (await api.rpc.chain.getHeader()).number.toNumber();
+
+  return new Promise<void>(async (resolve, _reject) => {
+    const unsub = await api.rpc.chain.subscribeNewHeads(async (header: any) => {
+      const queue = await (
+        await api.at(header.hash)
+      ).query.system.storageQueue();
+
+      for (let i = 0; i < queue.length; i++) {
+        for (let j = 0; j < queue[i][2].length; ++j) {
+          const tx = queue[i][2][j][1];
+          const tx_hash = blake2AsHex(tx);
+          if (myTxs.has(tx_hash)) {
+            exec_counter++;
+            myTxs.delete(tx_hash);
           }
+        }
+      }
```

Comment:
look into storage queue, check how many txs created by this execution thread has been enqueued, and submit same amount of new txs to keep the load steady